code/modules/mob/living/living.dm

```diff
@@ -1193,6 +1193,22 @@
 	mob_pickup(user)
 	return TRUE
 
+./mob/living/display_output(sound/S, mutable_appearance/vfx, text, turf/turf_source, vol as num)
```

Comment:
this should be handled by the datum, not the mob

Reply:
passing of information is handled on a mob level so that this functionality can be transferred to listener datums smoothly

Reply:
@Militaires I suppose, it feels like it's wrong to me though, datums should handle all the functionality of a thing inside them, and not be relying on mob or object behaviour.

if you want to change behaviours per type the answer is to sswap out the datums used, not have mob or object level callbacks.

/shrug I'm still undecided on this but if another maintianer wants to merge it that's on their head.

Reply:
Output datum actually represents the "sender" side of thing, it is in datum just for a unified system usage
Mob(later it will be special listener datum) represents the "reciever" side, for example if dude is high on drugs its not datum_output's job to replace its usual image with some bullshit, its something the mob(later listener datum) should handle
Technically you can both handle the image showing in output datum and then manually stop and replace it later in listener datum(currently just a mob), but first deciding whats going to be shown and only after than showing the end result is faster, thus its pushed to the later side, which is listener datum(currently mob)